monai/handlers/nvtx_handlers.py

```diff
@@ -0,0 +1,201 @@
+        """
+        event = event.upper()
+        event_prefix = {
+            "": "",
+            "ENGINE": "",
+            "EPOCH": "EPOCH_",
+            "ITERATION": "ITERATION_",
+            "BATCH": "GET_BATCH_",
+        }
+        return (
+            self.get_event(event_prefix[event] + "STARTED"),
+            self.get_event(event_prefix[event] + "COMPLETED"),
+        )
+
+    def get_event(self, event: Union[str, Events]) -> Events:
+        if isinstance(event, str):
+            return Events[event.upper()]
+        if isinstance(event, Events):
+            return event
+
+    def attach(self, engine: Engine) -> None:
+        """
+        Attach an NVTX Range to specific Ignite events
+        Args:
+            engine: Ignite Engine, it can be a trainer, validator or evaluator.
+        """
+        engine.add_event_handler(self.events[0], self.range_push)
+        engine.add_event_handler(self.events[1], self.range_pop)
+
+    def range_push(self):
+        self.depth = _nvtx.rangePushA(self.msg)
+
+    def range_pop(self):
+        _nvtx.rangePop()
+
+
+class RangePushHandler:
+    """
+    At a specific event, pushes a range onto a stack of nested range span.
+    Stores zero-based depth of the range that is started.
+
+    Args:
+        msg: ASCII message to associate with range
+    """
+
+    def __init__(self, event: Events, msg: Optional[str] = None) -> None:
+        if isinstance(event, str):
+            event = Events[event]
+        if msg is None:
+            msg = event.name
+        self.msg = msg
+        self.event = event
+        self.depth = None
+
+    def attach(self, engine: Engine) -> None:
+        """
+        Push an NVTX range at a specific Ignite event
+        Args:
+            engine: Ignite Engine, it can be a trainer, validator or evaluator.
+        """
+        engine.add_event_handler(self.event, self.range_push)
+
+    def range_push(self):
+        self.depth = _nvtx.rangePushA(self.msg)
+
+
+class RangePopHandler:
+    """
+    At a specific event, pop a previously pushed range.
+    Stores zero-based depth of the range that is started.
+
+    Args:
+        msg: ASCII message to associate with range
+    """
+
+    def __init__(self, event: Events) -> None:
+        if isinstance(event, str):
+            event = Events[event]
+        self.event = event
+
+    def attach(self, engine: Engine) -> None:
+        """
+        Pop an NVTX range at a specific Ignite event
+        Args:
+            engine: Ignite Engine, it can be a trainer, validator or evaluator.
+        """
+        engine.add_event_handler(self.event, self.range_pop)
+
+    def range_pop(self):
+        _nvtx.rangePop()
+
+
+class MarkHandler:
+    """
+    Mark an instantaneous event that occurred at some point.
+
+    Args:
+        msg: ASCII message to associate with range
+    """
+
+    def __init__(self, event: Events, msg: Optional[str] = None) -> None:
+        if isinstance(event, str):
```

Comment:
if event is `Event` and what will happen with `Events[event]`?

Reply:
Nothing, it's already an `Event` so it can be attached to. `self.event = event`

Reply:
I mean if `Events[event]` can work for `Event`, then no need to check `if isinstance(event, str)`, right?

Reply:
That's a great point! Thanks :)

Reply:
Change is done!